test/test-certs/create-certs.sh

```diff
@@ -28,23 +28,47 @@ openssl req \
     -passin pass:$SSL_SECRET \
     -passout pass:$SSL_SECRET
 
-for i in kafka kafka1 kafka2 schema-registry materialized producer postgres certuser
+# Create CA
+openssl req \
+    -x509 \
+    -days 36500 \
+    -newkey rsa:4096 \
+    -keyout secrets/ca-selective.key \
+    -out secrets/ca-selective.crt \
+    -sha256 \
+    -batch \
+    -subj "/CN=MZ RSA CA" \
+    -passin pass:$SSL_SECRET \
+    -passout pass:$SSL_SECRET
+
+
+for i in kafka kafka1 kafka2 schema-registry materialized producer postgres certuser materialized-kafka materialized-sr
```

Comment:
Hmm, I just noticed that there's two distinct things we're looping over here - things we need a cert and a jks for, and things we only need certs for. I think combining the two is what's leading to the complexity (ie. the need for a conditional within the loop, and the hardcoded "fixup" adding the additional certs to the appropriate jks after the loop's end). This might be a good opportunity to refactor so this is all a bit easier to maintain

One idea that might smooth things out:
1. Move the four openssl calls to a new function, eg. `create_cert(name, ca, cn)`
2. Create a new loop for things that only need certs, not jks
3. Restructure the jks loop so it always adds a service-specific cert alongside the CA to the truststore

In pseudocode, I think this would be something like:
```
for i in postgres certuser materialized producer
do
  create_cert($i, "ca", $i)
done

for i in kafka kafka1 kafka2 schema-registry
do
  create_cert($i, "ca", $i)
  create_cert("materialized-$i", "ca-selective", "materialized")
  # create keystore and import ca
  <two calls to keytool>
  # create truststore and import ca and selective cert
  <two calls to keytool>
done
```

Now everything that needs only certs doesn't get extraneous jks generated, and every java service that needs jks has a trusted CA and a trusted cert specific to the service, plus no special casing or conditionals for `materialized-kafka/-sr`.

What do you think?

Reply:
I was trying to create the minimal diff to get the test to pass. What do you think about me cleaning this up in a separate pr?

Reply:
Works for me!